### PR TITLE
Neo select component

### LINKED ISSUE #5852: Goodbye Buefy
## Rewrite [Buefy](https://buefy.org/documentation) components in `libs/ui` using [Oruga-ui](https://oruga.io/documentation/)
part of 
- #2690 

#### In Progress
- [x] replace buefy snackbar & toast #6744 

#### Done
- [x] `<b-progress />` => component or check Bulma #5895
- [x] `<b-carousel />` => `<NeoCarousel />` (Oruga [Carousel](https://oruga.io/components/Carousel.html))  #5884
- [x] `<b-carousel-item />` => `<NeoCarouselItem />` #5884
- [x] `<b-icon />` => `<NeoIcon />` #5858
- [x] `<b-modal />` => `<NeoModal />` #5883 
- [x] `<b-datepicker />` => `<NeoDatepicker />` (Oruga [Datepicker](https://oruga.io/components/Datepicker.html)) #5946
- [x] `<b-checkbox />` => `<NeoCheckbox />` #6000 
- [x] `<b-steps />` => `<NeoSteps />` (Oruga [Steps](https://oruga.io/components/Steps.html)) #5917
- [x] `<b-step-item />` => `<NeoStepItem />` #5917
- [x] `<b-switch />` => `<NeoSwitch />` (Oruga [Switch](https://oruga.io/components/Switch.html)) #5945
- [x] `<b-radio-button />` => `<NeoRadioButton />` (Oruga [Radio](https://oruga.io/components/radio.html)) #5976 
- [x] `<b-autocomplete />` => `<NeoAutocomplete />` (Oruga [Autocomplete](https://oruga.io/components/autocomplete.html)) #6018 
- [x] `<b-loading />` => `<NeoLoading />` (Oruga [Loading](https://oruga.io/components/Loading.html)) #5933
- [x] `<b-tooltip />` => #5856 
- [x] `<b-message />` => notification or Bulma [message](https://bulma.io/documentation/components/message/) - 3 results #6068
- [x] `<b-image />` => `` #5854
- [x] `<b-field />` => `<NeoField />` #5870 
- [x] `<b-input />` => `<NeoInput />` #5870
- [x] `<b-tag />` => `<NeoTag />` check Bulma [Tag](https://bulma.io/documentation/elements/tag/) - 3 results #6155 
- [x] `<b-table />` => `<NeoTable />` (Oruga [Table](https://oruga.io/components/Table.html)) - 8 results (related with #6110, #6144)
- [x] `<b-table-columns />` => `<NeoTableColumn />` - 87 results (related with #6110, #6144)
- [x] `<b-numberinput />` => `<NeoInput />` type=number??  - 3 results #6195 
- [x] `<b-taginput />` => `<NeoInputitems />` - 1 result #6183 
- [x] `<b-collapse />` => `<NeoCollapse />` (Oruga [Collapse](https://oruga.io/components/Collapse.html)) -  <s>13</s> 10 results #6249
- [x] `<b-dropdown />` => `<NeoDropdown />` - 3 results #6196
- [x] `<b-dropdown-item />` => `<NeoDropdownItem />` - 26 results #6196
- [x] `<b-slider />` => `<NeoSlider />` Oruga [slider](https://oruga.io/components/slider.html) - #6340
- [x] `<b-upload />` => `<NeoUpload />` Oruga [upload](https://oruga.io/components/upload.html) - #6341
- [x] `<b-pagination />` => `<NeoPagination />` Oruga [Pagination](https://oruga.io/components/pagination.html) #6366 
- [x] `<b-tabs />` => `<NeoTabs />` (Oruga [Tabs](https://oruga.io/components/Tabs.html))  - 9 results #6231 
- [x] `<b-tab-item />` => `<NeoTabItem />` -  15 results #6231 
- [x] `<b-navbar />` => [Bulma Navbar](https://bulma.io/documentation/components/navbar/) #6382 
- [x] `<b-navbar-item />` => custom html ??? #6382 
- [x] `<b-select />` => `<NeoSelect />` - 15 results #6200, #6338
- [x] `<b-button />` => `<NeoButton />` - <s>64</s> <s>47</s> <s>41</s> 35 results (make it the last one) #6437 #6449 
- [x] #6277 

lmk if you want to work on a specific component.

### PR DESCRIPTION
**Pull request by brano-hozza**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Related with #5852 
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=Ho4kw64mS5HmTCmGN8vCcQAazkrhqkqi7oyVcY64KjEPBcS)

#### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### <samp>🤖 Generated by Copilot at 8d677f1</samp>

The pull request replaces the `b-select` components from the Buefy library with the custom `NeoSelect` components from the UI library in various components of the NFT gallery. This improves the UI consistency and style across the application. The `NeoSelect` component is a mixin of the `OSelect` component from the Oruga library, with custom styles defined in the `NeoSelect.scss` file. The unit test file for the `BalanceInput` component is also updated to use the Oruga library.

### <samp>🤖 Generated by Copilot at 8d677f1</samp>

> _We rise from the ashes of Buefy_
> _We forge our own UI destiny_
> _We unleash the power of NeoSelect_
> _We make our components look perfect_

### CODE REVIEW

**Comment by roiLeo** [comment]
Screenshot 2023-06-27 at 08-35-32 Neo select component by brano-hozza · Pull Request #6338 · kodadot_nft-gallery

File: components/collection/unlockable/UnlockableLoader.vue [hunk 1]
```diff
@@ -72,7 +72,7 @@ const displaySeconds = computed(() => {
 
 const twitterText = computed(
   () =>
-    'Just minted an exclusive NFT with unlockable items on @Kodadot! 🎉 So excited to add this unique collectible to my collection. Don\'t miss your chance! \n\n https://kodadot.xyz/stmn/drops/free-drop'
+    "Just minted an exclusive NFT with unlockable items on @Kodadot! 🎉 So excited to add this unique collectible to my collection. Don't miss your chance! \n\n https://kodadot.xyz/stmn/drops/free-drop"
```
Comment (roiLeo) [inline components/collection/unlockable/UnlockableLoader.vue:75]:
```suggestion
    'Just minted an exclusive NFT with unlockable items on @Kodadot! 🎉 So excited to add this unique collectible to my collection. Don\'t miss your chance! \n\n https://kodadot.xyz/stmn/drops/free-drop'
```

**Review by roiLeo — APPROVED** [review]

**Comment by prury** [comment]
- Active and Hover effects on canary:

https://github.com/kodadot/nft-gallery/assets/36627808/4a94be0c-c5b3-4698-a3cb-9496d34b7ff9

Deployed version:

https://github.com/kodadot/nft-gallery/assets/36627808/467b3387-cc9f-4fa8-a9d0-cc4e06cf8228

**-------**
- User offers really big on mobile: 
image

Canary: 

image

i also summon @exezbcz to check if this changes checks out

@roiLeo do we use this Box Plot Chart anywhere?

**Comment by roiLeo** [comment]
> @roiLeo do we use this Box Plot Chart anywhere?

I don't see any usage in app, IMO it's safe to remove

**Comment by roiLeo** [comment]
Hello, can we make any progress on this issue? I can help if needed

**Comment by brano-hozza** [comment]
@roiLeo hey, I was too busy lately but I will try to look into that tonight

**Comment by brano-hozza** [comment]
Back to @prury comments, the comparison with canary, on canary there is still old design but I redesigned it by the figma prototype. And I am not sure how to get to user offers to check that thing :D

**Comment by vikiival** [comment]
> And I am not sure how to get to user offers to check that thing :D

@prury can you please share the link to the NFT?

**Comment by prury** [comment]
> > And I am not sure how to get to user offers to check that thing :D
> 
> @prury can you please share the link to the NFT?

NFT: https://deploy-preview-6338--koda-canary.netlify.app/bsx/gallery/3030053998-6

@brano-hozza in order to see the offers made tab, you must make an offer to any NFT on the BSX network, you can cancel it right afterwards, just to generate the registry

**Comment by exezbcz** [comment]
Guessing based on the designs (in simple mint figma), it should change to white stroke when active + the design of the dropdown under should be a bit different as well 
image

**Comment by roiLeo** [comment]
> Guessing based on the designs (in simple mint figma), it should change to white stroke when active + the design of the dropdown under should be a bit different as well 

Let's roll from now, we can open new issue for the remaining:
- arrow icon should be: #FFF on darkmode & #000 for lightmode
- replace `NeoSelect` with `NeoDropdown` on create pages

**Comment by exezbcz** [comment]
@roiLeo agree, lets go

**Comment by prury** [comment]
> > Guessing based on the designs (in simple mint figma), it should change to white stroke when active + the design of the dropdown under should be a bit different as well
> 
> Let's roll from now, we can open new issue for the remaining:
> 
>     * arrow icon should be: #FFF on darkmode & #000 for lightmode
> 
>     * replace `NeoSelect` with `NeoDropdown` on create pages

- offers made select size on mobile

File: components/collection/unlockable/UnlockableLoader.vue (hunk 1 above)
Comment (roiLeo) [inline components/collection/unlockable/UnlockableLoader.vue:75]:
```suggestion
    'Just minted an exclusive NFT with unlockable items on @Kodadot! 🎉 So excited to add this unique collectible to my collection. Don\'t miss your chance! \n\n https://kodadot.xyz/stmn/drops/free-drop'
```

**Review by roiLeo — APPROVED** [review]

**Review by preschian — APPROVED** [review]

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $55 @ 5.23 USD/DOT ~ 10.516 $DOT
🧗 [Ho4kw64mS5HmTCmGN8vCcQAazkrhqkqi7oyVcY64KjEPBcS](https://polkadot.subscan.io/account/Ho4kw64mS5HmTCmGN8vCcQAazkrhqkqi7oyVcY64KjEPBcS)
🔗 [0x07e23f7ea4a5f6a4aed0e019b3a8d50479962c850b5db6ee6b92fe9ba575dadb](https://polkadot.subscan.io/extrinsic/0x07e23f7ea4a5f6a4aed0e019b3a8d50479962c850b5db6ee6b92fe9ba575dadb)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**